Revert Revert "Add Infisical Docs" (#14762)
pulumi/docs: content/docs/esc/integrations/dynamic-secrets/infisical-secrets.md

---
title: infisical-secrets
title_tag: infisical-secrets Pulumi ESC provider

> meta_desc: The infisical-secrets Pulumi ESC Provider enables you to dynamically import secrets from Infisical Secrets into your environment.

Reviewer: ```suggestion
meta_desc: The infisical-secrets Pulumi ESC Provider enables you to dynamically import secrets from Infisical into your environment.
```

h1: infisical-secrets
menu:
  esc:
    identifier: infisical-secrets
    parent: esc-dynamic-secrets
    weight: 3
aliases:
  - /docs/pulumi-cloud/esc/providers/infisical-secrets/
  - /docs/esc/providers/infisical-secrets/
---

The `infisical-secrets` provider enables you to dynamically import Secrets from Infisical Secrets into
your Environment. The provider will return a map of names to Secrets.

## Example

```yaml
values:
  infisical:
    login:
      fn::open::infisical-login:
        oidc:
          identityId: aaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee
    secrets:
      fn::open::infisical-secrets:
        login: ${infisical.login}
        get:
          api-key:
            projectId: xxxxxxx-bbbb-cccc-dddd-eeeeeeeeeeee

>             environment: production

Reviewer: ```suggestion
            environment: prod
```

            secretKey: api-key
          app-secret:
            projectId: xxxxxxx-bbbb-cccc-dddd-eeeeeeeeeeee

>             environment: production

Reviewer: ```suggestion
            environment: dev
```

|---------------|--------|---------------------------------------------------------------------------------------------------------------------------|
| `siteUrl`     | string | [Optional] - The base URL of the Infisical instance you authenticated to. May be omitted if default US instance was used. |
| `accessToken` | string | The access token to use for authentication.                                                                               |

### InfisicalSecretsGet

| Property      | Type   | Description                                                                                                                                                                                         |
|---------------|--------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `projectId`   | string | The projectId of the project the secret lives in. In the Infisical web app, navigate to your Secrets project, click on *Project Settings*, and click the *Copy Project ID* button.                  |

> | `environment` | string | The environment slug of the environment the secret lives in. In the Infisical web app, navigate to your Secrets project, click on *Project Settings*, and find the slug in the *Environments* list. |

Reviewer: ```suggestion
| `environment` | string | The environment slug of the environment the secret lives in. In the Infisical web app, navigate to your Secrets project, click on *Project Settings*, and find the slug in the *Environments* list. Default values are `dev`, `staging`, and `prod` |
```

| `secretKey`   | string | The name of the secret to import.                                                                                                                                                                   |

> | `secretPath`  | string | [Optional] - The path inside the environment the secret lives in.                                                                                                                                   |

Reviewer: ```suggestion
| `secretPath`  | string | [Optional] - The path inside the environment where the secret lives. For example, if your secret `dbPassword' lives within `DatabaseDetails` folder. The path would be `\DatabaseDetails`                                                                                                                                   |
```
